cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResources.java

```diff
@@ -0,0 +1,119 @@
+        }
+        String sourceVaultId = String.format("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.KeyVault/vaults/%s",
+                azureClient.getCurrentSubscription().subscriptionId(), vaultResourceGroupName, vaultName);
+
+        CreatedDiskEncryptionSet diskEncryptionSet = getOrCreateDiskEncryptionSetOnCloud(
+                azureClient,
+                desResourceGroupName,
+                sourceVaultId,
+                diskEncryptionSetCreationRequest);
+        azureClient.grantKeyVaultAccessPolicyToServicePrincipal(vaultResourceGroupName, vaultName,
+                diskEncryptionSet.getDiskEncryptionSetPrincipalId());
+        return diskEncryptionSet;
+    }
+
+    private CreatedDiskEncryptionSet getOrCreateDiskEncryptionSetOnCloud(AzureClient azureClient, String desResourceGroupName,
+            String sourceVaultId, DiskEncryptionSetCreationRequest diskEncryptionSetCreationRequest) {
+        String diskEncryptionSetName = azureUtils.generateDesNameByNameAndId(
+                String.format("%s-DES-", diskEncryptionSetCreationRequest.getEnvironmentName()),
+                diskEncryptionSetCreationRequest.getId());
+        LOGGER.info("Checking if Disk Encryption Set {} exists on cloud Azure", diskEncryptionSetName);
+        DiskEncryptionSetInner createdSet = azureClient.getDiskEncryptionSet(desResourceGroupName, diskEncryptionSetName);
+        if (createdSet == null) {
+            LOGGER.info("Creating Disk Encryption Set {}", diskEncryptionSetName);
+            createdSet = azureClient.createDiskEncryptionSet(diskEncryptionSetName, diskEncryptionSetCreationRequest.getEncryptionKeyUrl(),
+                    diskEncryptionSetCreationRequest.getRegion().getRegionName(), desResourceGroupName, sourceVaultId,
+                    diskEncryptionSetCreationRequest.getTags());
+        }
+        if (createdSet != null) {
+            CloudResource desCloudResource = CloudResource.builder()
+                    .name(diskEncryptionSetName)
+                    .type(AZURE_DISK_ENCRYPTION_SET)
+                    .reference(createdSet.id())
+                    .status(CommonStatus.CREATED)
+                    .build();
+            persistenceNotifier.notifyAllocation(desCloudResource, diskEncryptionSetCreationRequest.getCloudContext());
+
+            return new CreatedDiskEncryptionSet.Builder()
+                    .withDiskEncryptionSetId(createdSet.id())
+                    .withDiskEncryptionSetPrincipalId(createdSet.identity().principalId())
+                    .withDiskEncryptionSetLocation(createdSet.location())
+                    .withDiskEncryptionSetName(createdSet.name())
+                    .withTags(createdSet.getTags())
+                    .withDiskEncryptionSetResourceGroup(desResourceGroupName)
+                    .build();
+        } else {
+            throw new CloudConnectorException("Failed to create Disk Encryption Set.");
+        }
```

Comment:
Azure SDK call either returns the object(filled(with DES id and other values)) or throws `IllegalArgumentException`.

Reply:
OK, so this is mainly for the test case. We can leave this behavior, but it would be a really bad Azure SDK design fault if we ever encountered `null` in practice instead of getting an exception.